File: 20-ValidParentheses/20-ValidParentheses.java
// Last updated: 7/14/2026, 2:17:18 PM
class Solution {
    public boolean isValid(String s) {
        char arr[]=new char[s.length()];
        int ind=-1;
        for(char ch:s.toCharArray()){
            if (ch=='('){
                arr[++ind]=')';
            }
            else if(ch=='{'){
                arr[++ind]='}';
            }
            else if(ch=='['){
                arr[++ind]=']';

            }
            else if(ind==-1){
                return false;
            }
            else if(arr[ind--]!=ch){
                return false;
            }
        }
        return ind==-1;
       }
}

File: 20-ValidParentheses/20-ValidParentheses.py
# Last updated: 7/14/2026, 2:17:13 PM
class Solution:
    def isValid(self, s: str) -> bool:
        st=[]
        d={ "(":")","[":"]","{":"}" }

        for i in s:
            if i in d.keys():
                st.append(i)
            else:
                if st==[]:
                    return False
                else:
                    if d[st[-1]]==i:
                        st.pop()
                    else:
                        return False
        
        if st==[]:
            return True
        else:
            return False
